src/ascent/runtimes/expressions/ascent_blueprint_architect.hpp

```diff
@@ -91,51 +116,92 @@ void ASCENT_API binning_mesh(const conduit::Node &binning,
                              conduit::Node &mesh,
                              const std::string field_name = "");
 
+void ASCENT_API binning_mesh2(const conduit::Node &binning, conduit::Node &mesh);
```

Comment:
- [ ] make decision on binning vs binning2

Reply:
I decided to re-write the binning for two reasons. This is all from memory so i might be slightly wrong. 1) The results were actually incorrect. 2) The parallel implementation had race conditions.

Reply:
There are actual 3 versions of binning. Maybe we could chat about the right path?

My plan right now is to keep the old, and then work on a new RAJA version. Keeping old since the unit tests still pass, and too many moving pieces stalled me my last attempt to merge.  Need to get overall RAJA exprs in then plot a path forward.

Reply:
@mclarsen Using RAJA binning will be future work, but I solved the other reduction mysteries. Thanks for your help discussing this!